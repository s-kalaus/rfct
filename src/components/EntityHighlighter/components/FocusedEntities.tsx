import React from 'react';
import { IEntity } from '~/interfaces/entity';
import { FocusedEntity } from './FocusedEntity';

export interface IProps {
  text: string;
  entities: IEntity[];
  onDelete: (entity: IEntity) => void;
}

export const FocusedEntities = ({ entities, text, onDelete }: IProps) => (
  <>
    {entities.map((entity) => (
      <FocusedEntity
        text={text}
        entity={entity}
        key={JSON.stringify(entity)}
        onDelete={onDelete}
      />
    ))}
  </>
);
