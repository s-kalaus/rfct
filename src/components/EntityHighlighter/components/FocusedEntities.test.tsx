import React from 'react';
import { render } from '@testing-library/react';
import { FocusedEntities } from './FocusedEntities';

describe('FocusedEntities', () => {
  const text = 'Text';
  const entities = [{ selectionStart: 0, selectionEnd: 0, label: 'Label' }];
  const onDelete = jest.fn();

  test('renders component', () => {
    const { container } = render(
      <FocusedEntities text={text} entities={entities} onDelete={onDelete} />
    );
    expect(container).toMatchSnapshot();
  });
});
