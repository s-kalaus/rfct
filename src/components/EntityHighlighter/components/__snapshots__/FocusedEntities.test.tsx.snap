// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusedEntities renders component 1`] = `
<div>
  <div
    style="word-break: break-word;"
  >
    
     (Label)
    <button
      data-testid="button"
      style="border: 0px none; cursor: pointer; background-color: transparent;"
    >
      <span
        aria-label="Delete"
        role="img"
      >
        🗑️
      </span>
    </button>
  </div>
</div>
`;
